tests/core/test_single_derived_signal.py

```diff
@@ -0,0 +1,161 @@
+import asyncio
+from unittest.mock import call
+
+import pytest
+
+from ophyd_async.core import (
+    AsyncStatus,
+    Device,
+    DeviceVector,
+    StandardReadable,
+    StrictEnum,
+    derived_signal_r,
+    derived_signal_rw,
+    derived_signal_w,
+    soft_signal_rw,
+)
+from ophyd_async.testing import (
+    assert_describe_signal,
+    assert_reading,
+    assert_value,
+    get_mock,
+)
+
+
+class BeamstopPosition(StrictEnum):
+    IN_POSITION = "In position"
+    OUT_OF_POSITION = "Out of position"
+
+
+class Beamstop(Device):
+    def __init__(self, name=""):
+        # Raw signals
+        self.x = soft_signal_rw(float)
+        self.y = soft_signal_rw(float)
+        # Derived signals
+        self.position = derived_signal_r(self._get_position, x=self.x, y=self.y)
+        super().__init__(name=name)
+
+    def _get_position(self, x: float, y: float) -> BeamstopPosition:
+        if abs(x) < 1 and abs(y) < 2:
+            return BeamstopPosition.IN_POSITION
+        else:
+            return BeamstopPosition.OUT_OF_POSITION
+
+
+class Positioner(Device):
+    def __init__(self, name=""):
+        # Raw signals
+        self.x = soft_signal_rw(float)
+        self.y = soft_signal_rw(float)
+        # Derived signals
+        self.position = derived_signal_rw(
+            self._get_position, self.set, x=self.x, y=self.y
+        )
+        super().__init__(name=name)
+
+    def _get_position(self, x: float, y: float) -> BeamstopPosition:
```

Comment:
Nit: This is probably something we discussed already in person but it's a little annoying the type system won't catch this having incorrect arguments. I assume you've thought about it and there's no clean way round it. We can live with it

Reply:
3 days of head banging and I can't see a way of convincing the type system to check this...

I will make another ticket to add runtime checking of these at init time though